includes/class-sensei-frontend.php

```diff
@@ -1067,72 +1067,6 @@ public function sensei_course_category_main_content() {
 
 	} // End sensei_course_category_main_content()
 
-	public function sensei_login_form() {
-		?>
-		<div id="my-courses">
-			<?php Sensei()->notices->maybe_print_notices(); ?>
-			<div class="col2-set" id="customer_login">
-
-				<div class="col-1">
-					<?php
-					// output the actual form markup
-                    Sensei_Templates::get_template( 'user/login-form.php');
-					?>
-				</div>
-
-			<?php
-			if ( get_option('users_can_register') ) {
-
-				// get current url
-				$action_url = get_permalink();
-
-				?>
-
-				<div class="col-2">
-					<h2><?php _e( 'Register', 'woothemes-sensei' ); ?></h2>
-
-					<form method="post" class="register"  action="<?php echo esc_url( $action_url ); ?>" >
```

Comment:
What do you think is a preferable way to fix this issue?

I tried modifying the pluggable `wp_new_user_notification` to contain `redirect_to=wp_get_referer()` but that didn't work as they get another redirection after setting their password correctly.

Other thing I thought maybe we could try to mark the user in the DB somehow, so after they log in they get redirected properly...

The third thing that came to my mind was to use cookies, but I don't like using them :)